### ISSUE TITLE
Feature Requests & To-Do

### ISSUE THREAD

**Issue by SceptreOfficial** [SceptreOfficial/Simplex-Support-Services#12]
Feel free to comment on this issue about any feature requests or tweaks.

**Comment by NortonDK** [comment]
Make helicopter pickups able to land on IR strobes, chemlights

**Comment by SceptreOfficial** [comment]
Helicopters will automatically switch to detect IR strobes at night.

**Comment by EpsilonCommunity** [comment]
Suppress Area Command?

**Comment by SceptreOfficial** [comment]
> Suppress Area Order?

For what service? More details plz

**Comment by Burbonizer** [comment]
Request support drops, such as ammo crates or vehicles.

**Comment by NZF-JDWang** [comment]
First of thanks for adding the item restrictions and milsim modes I requested, they're awesome.

Now I have 1 more use case I'd like you to consider.

I have players out in the field who can only call chopper transport when they have a long range radio pack, which is perfect. But when someone respawns at base it would be cool if they could board the chopper and reinsert without needing the long range radio. 

Basically I want to be able to simulate communication lines with the pilot, so if you're NOT in the chopper you need the radio (which you've already done thank you) but if you ARE in the chopper then there's no restriction and you can order the pilot to reinsert you.

With ALiVE transports they have a "talk to the pilot" action for anyone who boards the chopper.

Thanks again for the great mod.

**Comment by SceptreOfficial** [comment]
There already is a transport action on the transport vehicles, however it only shows up for assigned units at the moment... plus there is a JIP issue in the current version. The next update will fix the JIP issue and have the option for anyone that's in/near the transport to 'talk to the pilot'

**Comment by NZF-JDWang** [comment]
Awesome thank you.
Does the JIP issue still arise when you name the players rather than just syncing them?

This is very quickly becoming a must have mod, thanks for all the work you're putting into it.

**Comment by SceptreOfficial** [comment]
Yeah, np. 
It's not related to assignment, its that the action on the vehicle is only added once globally. It doesn't account for any players that join after that.

**Comment by NZF-JDWang** [comment]
How about a slingload service for heavy lift choppers?
You could send the chopper to a location and when it gets there maybe you get a populated list of containers or slingloadable vehicles in the area? You select the item the chopper hooks it and then you give it the delivery location which it flies to and unhooks.

Would be a good way to run resupply for milsim groups without pilots

**Comment by SceptreOfficial** [comment]
Sling loading was just included with the latest update. Just how you described, too. Objects have to be sling loadable by native standards though.

**Comment by ProXeN** [comment]
It would be great to add the option to reposition drones/gunships.

**Comment by NZF-JDWang** [comment]
Can you add more than one support type per vehicle? So for instance thew UH-60DAP blackhawks can act as both transport and CAS.

**Comment by SceptreOfficial** [comment]
No. I assume you just want the search and destroy function on transport helps? I could make that an extra option.

**Comment by NZF-JDWang** [comment]
That would work, although the ability to do more targeted attack would be even better.

Some way make the use of multirole choppers like the Blackhawk DAP or Russian Hinds etc would be great. They're both gunships which can carry troops

**Comment by DarkLab-s** [comment]
In SSS settings would it be possible to customize the hover height cap?

Say you wanted to use the "lock" hover at 2000m and allow for walk out para jumping. 

Right now it is locked at 40m regardless of what you type into the cell.

**Comment by NZF-JDWang** [comment]
One more request, can you please add the ACE interactions to Zeus as well please.
Unless I'm missing something I have to come out of Zeus in order to request supports, it would be great to be able to access them while still in Zeus.

I believe you'd just have to use `ace_interact_menu_fnc_addActionToZeus`

**Comment by SceptreOfficial** [comment]
Thanks for the feedback, guys. I'll try to get some of these things done after I finish converting everything over for language support.

@NZF-JDWang 
> That would work, although the ability to do more targeted attack would be even better.

Like a gun run on a specific target? I'm not sure if there is an easy way to do it natively and reliably. I know Achilles had a module that worked through some velocity/position manipulation. If I have to, I can look at it and get some ideas.

**Comment by NZF-JDWang** [comment]
Yeah I know with ALiVE supports I can set up a Hind, Mi-8 or UH-60 DAP to work as a transport chopper but then also call it in for a gun or rocket/missile run.

Essentially it would be like linking the chopper to both the CAS module and the transport module.

**Comment by NZF-JDWang** [comment]
Yet another feature request would be "no fire" areas that the mission maker can define and CAS/Artillery etc will not accept missions to fire on that area.

Eg the mission maker could put down markers around all mosques, name the markers and then CAS/Artillery will refuse fire missions inside the marker areas

**Comment by SceptreOfficial** [comment]
> 
> 
> Yet another feature request would be "no fire" areas that the mission maker can define and CAS/Artillery etc will not accept missions to fire on that area.
> 
> Eg the mission maker could put down markers around all mosques, name the markers and then CAS/Artillery will refuse fire missions inside the marker areas

Naughty playerbase I assume?
I can probably add a "request approval condition" or something like that. It would require some scripting knowledge from mission makers, but it would just be a simple position/inArea check.
Something like: `!(_position inArea "NoArtilleryMarker)"`

Thoughts?

**Comment by NZF-JDWang** [comment]
Yeah if we can define multiple areas that would be great.

Not so much a naughty playerbase, more over enthusiastic :-P 
I just want them to think before bombing haha

**Comment by SceptreOfficial** [comment]
Yeah, with it just being a code condition, you could come up with any kind of condition required.
For multiple markers you could use this:
`{_position inArea _x} count ["marker1","marker2","marker3"] == 0`

Basically if condition is true, fire mission is a go. If false, it gets denied.
Someone else suggested something similar for transport requests because he wanted to limit the distance that a request could be made.

**Comment by NZF-JDWang** [comment]
Yeah that would be cool too, being able to simulate being out of radio distance or "no fly zones" always makes things more interesting.

Really is a great mod you've made, thanks for all the work you've done and for continuing to support it with these feature requests

**Comment by EpsilonCommunity** [comment]
Yes, we use this mod all the time. Our JTAC/RTO loves this over commanding real people ;)

**Comment by NZF-JDWang** [comment]
Quick question...

Will the cooldown fields accept `random [60, 120, 360]` type inputs? 
Just thought it would be cool to add a little randomization in there.

**Comment by SceptreOfficial** [comment]
No, but you could use a CBA event to modify the cooldown on request submission. I already have some events added, but I probably should make sure they are ironed out before anyone starts using them. I'm going to release a small update to address a couple things very soon. I'll post an example when that is published.

**Comment by KiloRomeo15** [comment]
hi dude,

Your mod was brought to my attention by the Alive community and so far im impressed,

One thing, would it be possible to allow a "time on target" for CAS aircraft, 

Its all well and good my A10 flying over to BRRRRRTTTTT some poor dudes house, but then the pilot goes straight home for tea and medals, and I cant retask him for extra fires until hes flown home, done the ceremony and had a nap :D 

Many thanks 

Keep up the good work

**Comment by SceptreOfficial** [comment]
Getting fixed-wing AI to re-align and fire on a new target is completely unreliable, so that's why the way it works now is all completely manual velocity/position control.

**Comment by EpsilonCommunity** [comment]
Might I suggest another way, having a X requests at a time. Then a cooldown?

**Comment by NZF-JDWang** [comment]
Just checked the code but couldn't tell for sure, but can you make it so when you create a logistics point the items spawn on the module location. 

That way we can set a specific spot for items to spawn and they won't appear the other side of a wall or something.

Thanks

**Comment by GoldJohnKing** [comment]
The Transport module do not restore heli's skin/variant, loadout and crew when respawning it, request to support this.

**Comment by GoldJohnKing** [comment]
Feature Request: Add Multilingual Support.

**Comment by KiloRomeo15** [comment]
For Artillery, would it be at all possible to have a "REPEAT" for fire missions ?

**Comment by SceptreOfficial** [comment]
> For Artillery, would it be at all possible to have a "REPEAT" for fire missions ?
What do you mean by this?

**Comment by KiloRomeo15** [comment]
> > For Artillery, would it be at all possible to have a "REPEAT" for fire missions ?
> 
> What do you mean by this?

As in, repeat the last fire mission using the same location 

so I dont have to put the same grids in again 

if that makes any sense ?

**Comment by NZF-JDWang** [comment]
Is there, or can we have a way to make the helo's automagically RTB after they've fastroped everyone?
Sometime I just want to use Simplex for mission insertion and it would solve the problem of the players needing control over the helo in order to send it RTB after they fastropped out.

**Comment by CommandoCat** [comment]
Oooo I've got a good one! How about another CAS type: Carpet Bombing!

Works same as CAS but instead of the plane shooting guns/missiles/rockets at the target point, plane will fly horizontally past and will instead drop a stick of bombs on the target point/laser target (player can choose how many, spread of bombs, type of bomb can be added to the script in editor)

**Comment by NZF-JDWang** [comment]
This could be a little outside the scope of the mod, but I figure you have 90% of the code for it already, but it'd be great if we could have some modules for autonomous airstrikes. There's very few scripts etc for AI CAS runs and the BI module only works with vanilla aircraft.

Eg in the editor I place the module and give it the aircraft classname, weapon etc like you already do for the normal CAS aircraft, only this time I rotate the module to indicate ingress direction and now when I sync it to a trigger, when that trigger goes off the aircraft spawns (could also add a spawn distance parameter) and strikes the module location.  

Could do the same for artillery etc. 

Would add even more flexibility for mission makers.

**Comment by SceptreOfficial** [comment]
> Is there, or can we have a way to make the helo's automagically RTB after they've fastroped everyone?
> Sometime I just want to use Simplex for mission insertion and it would solve the problem of the players needing control over the helo in order to send it RTB after they fastropped out.

I thought I replied to this already, but it's already going to be added in the next update.

> Oooo I've got a good one! How about another CAS type: Carpet Bombing!
> 
> Works same as CAS but instead of the plane shooting guns/missiles/rockets at the target point, plane will fly horizontally past and will instead drop a stick of bombs on the target point/laser target (player can choose how many, spread of bombs, type of bomb can be added to the script in editor)

More CAS parameters is on the to-do list and I'm wanting to add it for the next major update.

> This could be a little outside the scope of the mod, but I figure you have 90% of the code for it already, but it'd be great if we could have some modules for autonomous airstrikes. There's very few scripts etc for AI CAS runs and the BI module only works with vanilla aircraft.
> 
> Eg in the editor I place the module and give it the aircraft classname, weapon etc like you already do for the normal CAS aircraft, only this time I rotate the module to indicate ingress direction and now when I sync it to a trigger, when that trigger goes off the aircraft spawns (could also add a spawn distance parameter) and strikes the module location.
> 
> Could do the same for artillery etc.
> 
> Would add even more flexibility for mission makers.

Possibly. If anything I will probably make some generic functions that would be easy to use. I've already done this for artillery to fix the inconsistency of the vanilla commands. But, not sure about specific modules.

**Comment by D0M51957** [comment]
Feature requests: Option to change the flyinheight ceiling up to 10,000m ASL for all fixed wing aircraft and 5000m ASL for all rotary wing aircraft. Currently maxed out at 2000m ASL.

Option to increase loiter radius up to 10km for all fixed wing aircraft. Currently maxed out at 1500m.

This will add realism and immersion for units/players interested in high altitude parachute operations. Thank you

**Comment by rco-titus** [comment]
Feature Request: Option to delete vehicles or objects within a radius of a logistics station, Incase of accidental spawning of a wrong vehicle or object or spawning too many. Instead of needing someone in zues to clean up unwanted vehicles that are cluttering the area, when opening the logistics station you could select an option to clear the area

**Comment by SceptreOfficial** [comment]
> Feature requests: Option to change the flyinheight ceiling up to 10,000m ASL for all fixed wing aircraft and 5000m ASL for all rotary wing aircraft. Currently maxed out at 2000m ASL.
> 
> Option to increase loiter radius up to 10km for all fixed wing aircraft. Currently maxed out at 1500m.
> 
> This will add realism and immersion for units/players interested in high altitude parachute operations. Thank you

Done

**Comment by SceptreOfficial** [comment]
Most of the requests here have been implemented. If there was anything missed or if you have a new idea, please open a new ticket and label it as a 'feature request'.
